packages/edit-site/src/components/editor/index.js

```diff
@@ -210,6 +214,12 @@ export default function Editor( { isLoading, onClick } ) {
 		[ entitiesSavedStatesCallback ]
 	);
 
+	useEffect( () => {
+		if ( isPublishSidebarOpened && ! isEditMode ) {
+			closePublishSidebar();
+		}
+	}, [ isEditMode, isPublishSidebarOpened, closePublishSidebar ] );
```

Comment:
Do we need to do that using a hook? Can't we just not render the sidebar?

Reply:
I think we do, because if we conditionally render the sidebar based on the edit mode every time we go again in that mode, the publish sidebar is rendered open(editor store state). This also means that if we:
1. have opened the sidebar
2. go view mode
3. navigate to different page
4. go edit mode
The sidebar still renders opened..

Reply:
Can we do it in `setCanvasMode` then? There's already a bunch of similar things there.

Reply:
Oh, yeah, it seems a better place.. Will try that.